### ISSUE TITLE
No monitoring data in cases of Multi-dc and ipv6 

### ISSUE THREAD

**Issue by fruch** [scylladb/scylla-cluster-tests#8229]
## Packages

Scylla version: `6.2.0~dev-20240802.39b49a41cc5d` with build-id `b5677630be86ba0391852836a26ee357a4912e5c`

Kernel Version: `6.8.0-1012-aws`

## Issue description

- [x] This issue is a regression.
- [ ] It is unknown if this issue is a regression.

on a multi-dc run with IPv6, we can't see any monitoring data 

## How frequently does it reproduce?

seem like it's reproducing every week, on the tier1 tests

## Installation details

Cluster size: 12 nodes (i3en.2xlarge)

Scylla Nodes used in this run:
- parallel-topology-schema-changes-mu-db-node-7df0cd5b-9 (3.8.21.186 | 2a05:d01c:0964:7d02:2613:283a:5c27:1a9a) (shards: 7)
- parallel-topology-schema-changes-mu-db-node-7df0cd5b-8 (35.178.225.28 | 2a05:d01c:0964:7d02:fed4:969f:996d:d9ac) (shards: 7)
- parallel-topology-schema-changes-mu-db-node-7df0cd5b-7 (13.40.145.254 | 2a05:d01c:0964:7d02:cd51:b1fc:7a73:2d46) (shards: 7)
- parallel-topology-schema-changes-mu-db-node-7df0cd5b-6 (3.253.34.160 | 2a05:d018:12e3:f002:d2dc:0eb6:1525:6b9f) (shards: 7)
- parallel-topology-schema-changes-mu-db-node-7df0cd5b-5 (3.252.84.184 | 2a05:d018:12e3:f002:7573:2a0b:6d1d:78d6) (shards: 7)
- parallel-topology-schema-changes-mu-db-node-7df0cd5b-4 (54.170.243.36 | 2a05:d018:12e3:f002:7283:b872:84cf:67b0) (shards: 7)
- parallel-topology-schema-changes-mu-db-node-7df0cd5b-3 (3.253.24.212 | 2a05:d018:12e3:f002:c659:bf20:2f71:c7b7) (shards: 7)
- parallel-topology-schema-changes-mu-db-node-7df0cd5b-22 (18.134.13.159 | 2a05:d01c:0964:7d02:39f3:aafb:e517:5ba7) (shards: 7)
- parallel-topology-schema-changes-mu-db-node-7df0cd5b-21 (3.10.138.132 | 2a05:d01c:0964:7d02:f393:91ec:87d7:ae28) (shards: -1)
- parallel-topology-schema-changes-mu-db-node-7df0cd5b-20 (3.249.197.3 | 2a05:d018:12e3:f002:6607:e5fb:d708:0061) (shards: 7)
- parallel-topology-schema-changes-mu-db-node-7df0cd5b-2 (54.154.154.66 | 2a05:d018:12e3:f002:c21d:c891:7c1b:b3ed) (shards: 7)
- parallel-topology-schema-changes-mu-db-node-7df0cd5b-19 (13.40.13.243 | 2a05:d01c:0964:7d02:255e:b006:2f79:d554) (shards: 7)
- parallel-topology-schema-changes-mu-db-node-7df0cd5b-18 (18.170.25.171 | 2a05:d01c:0964:7d02:50bc:806b:17b6:b240) (shards: 7)
- parallel-topology-schema-changes-mu-db-node-7df0cd5b-17 (34.244.195.182 | 2a05:d018:12e3:f002:2a49:a911:af6c:9b82) (shards: 7)
- parallel-topology-schema-changes-mu-db-node-7df0cd5b-16 (18.201.61.126 | 2a05:d018:12e3:f002:77d7:f6fe:73f6:1709) (shards: 7)
- parallel-topology-schema-changes-mu-db-node-7df0cd5b-15 (3.255.97.145 | 2a05:d018:12e3:f002:3554:f108:9309:4b7c) (shards: 7)
- parallel-topology-schema-changes-mu-db-node-7df0cd5b-14 (52.213.153.230 | 2a05:d018:12e3:f002:034f:261d:fc63:b6f2) (shards: 7)
- parallel-topology-schema-changes-mu-db-node-7df0cd5b-13 (13.40.161.121 | 2a05:d01c:0964:7d02:7d1d:f3cd:9f2a:f009) (shards: 7)
- parallel-topology-schema-changes-mu-db-node-7df0cd5b-12 (18.132.212.159 | 2a05:d01c:0964:7d02:7326:936c:5668:d5a7) (shards: 7)
- parallel-topology-schema-changes-mu-db-node-7df0cd5b-11 (3.10.144.14 | 2a05:d01c:0964:7d02:b0e5:f2a0:0d56:8f30) (shards: 7)
- parallel-topology-schema-changes-mu-db-node-7df0cd5b-10 (18.171.227.56 | 2a05:d01c:0964:7d02:36c1:a9c4:0893:255b) (shards: 7)
- parallel-topology-schema-changes-mu-db-node-7df0cd5b-1 (34.252.110.114 | 2a05:d018:12e3:f002:1627:1179:2975:ab34) (shards: 7)

OS / Image: `ami-00355186dfc821610 ami-0a8ad1e3e6b762c97` (aws: undefined_region)

Test: `longevity-multidc-schema-topology-changes-12h-test`
Test id: `7df0cd5b-d6fa-4a67-b476-52d54cad952e`
Test name: `scylla-master/tier1/longevity-multidc-schema-topology-changes-12h-test`
Test method: `longevity_test.LongevityTest.test_custom_time`
Test config file(s):

- [longevity-multidc-parallel-topology-schema-changes-12h.yaml](https://github.com/scylladb/scylla-cluster-tests/blob/3eefb3c644ddb52172bda2b0caf4fa4c36ca5fa6/test-cases/longevity/longevity-multidc-parallel-topology-schema-changes-12h.yaml)

<details>
<summary>
Logs and commands
</summary>

- Restore Monitor Stack command: `$ hydra investigate show-monitor 7df0cd5b-d6fa-4a67-b476-52d54cad952e`
- Restore monitor on AWS instance using [Jenkins job](https://jenkins.scylladb.com/view/QA/job/QA-tools/job/hydra-show-monitor/parambuild/?test_id=7df0cd5b-d6fa-4a67-b476-52d54cad952e)
- Show all stored logs command: `$ hydra investigate show-logs 7df0cd5b-d6fa-4a67-b476-52d54cad952e`

## Logs:
- **db-cluster-7df0cd5b.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/7df0cd5b-d6fa-4a67-b476-52d54cad952e/20240804_034710/db-cluster-7df0cd5b.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/7df0cd5b-d6fa-4a67-b476-52d54cad952e/20240804_034710/db-cluster-7df0cd5b.tar.gz)
- **sct-runner-events-7df0cd5b.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/7df0cd5b-d6fa-4a67-b476-52d54cad952e/20240804_034710/sct-runner-events-7df0cd5b.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/7df0cd5b-d6fa-4a67-b476-52d54cad952e/20240804_034710/sct-runner-events-7df0cd5b.tar.gz)
- **sct-7df0cd5b.log.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/7df0cd5b-d6fa-4a67-b476-52d54cad952e/20240804_034710/sct-7df0cd5b.log.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/7df0cd5b-d6fa-4a67-b476-52d54cad952e/20240804_034710/sct-7df0cd5b.log.tar.gz)
- **loader-set-7df0cd5b.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/7df0cd5b-d6fa-4a67-b476-52d54cad952e/20240804_034710/loader-set-7df0cd5b.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/7df0cd5b-d6fa-4a67-b476-52d54cad952e/20240804_034710/loader-set-7df0cd5b.tar.gz)
- **monitor-set-7df0cd5b.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/7df0cd5b-d6fa-4a67-b476-52d54cad952e/20240804_034710/monitor-set-7df0cd5b.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/7df0cd5b-d6fa-4a67-b476-52d54cad952e/20240804_034710/monitor-set-7df0cd5b.tar.gz)

[Jenkins job URL](https://jenkins.scylladb.com/job/scylla-master/job/tier1/job/longevity-multidc-schema-topology-changes-12h-test/29/)
[Argus](https://argus.scylladb.com/test/428b2fb3-e9ac-4276-b71a-9bf4b45a431a/runs?additionalRuns[]=7df0cd5b-d6fa-4a67-b476-52d54cad952e)
</details>

**Comment by fruch** [comment]
seems like we are patching the wrong file for quite some time, `/usr/lib/systemd/system/node-exporter.service` doesn't exists.
```
< t:2024-08-03 08:48:27,772 f:remote_base.py  l:521  c:RemoteLibSSH2CmdRunner p:DEBUG > <2a05:d01c:0964:7d02:39f3:aafb:e517:5ba7>: Running command "grep "web.listen-address" /usr/lib/systemd/system/node-exporter.service"...
< t:2024-08-03 08:48:28,345 f:base.py         l:231  c:RemoteLibSSH2CmdRunner p:DEBUG > <2a05:d01c:0964:7d02:39f3:aafb:e517:5ba7>: grep: /usr/lib/systemd/system/node-exporter.service: No such file or directory
```

and scylla node explorer might no be listing on the ipv6 address

**Comment by fruch** [comment]
> seems like we are patching the wrong file for quite some time, `/usr/lib/systemd/system/node-exporter.service` doesn't exists.
> 
> ```
> < t:2024-08-03 08:48:27,772 f:remote_base.py  l:521  c:RemoteLibSSH2CmdRunner p:DEBUG > <2a05:d01c:0964:7d02:39f3:aafb:e517:5ba7>: Running command "grep "web.listen-address" /usr/lib/systemd/system/node-exporter.service"...
> < t:2024-08-03 08:48:28,345 f:base.py         l:231  c:RemoteLibSSH2CmdRunner p:DEBUG > <2a05:d01c:0964:7d02:39f3:aafb:e517:5ba7>: grep: /usr/lib/systemd/system/node-exporter.service: No such file or directory
> ```
> 
> and scylla node explorer might no be listing on the ipv6 address

seems like a red herring, it was failing also on working setups.

seems like it's broken since we move to ubuntu 2404

**Comment by fruch** [comment]
trying a short longevity to investigate it:
https://argus.scylladb.com/test/813c3d10-2e69-4322-943f-7356ead5fa33/runs?additionalRuns[]=a7618452-bbff-46f8-8445-b24def1e7a70

**Comment by fruch** [comment]
nailed it to issue in monitoring itself

https://github.com/scylladb/scylla-monitoring/pull/2372

**Comment by fruch** [comment]
it's fixed in monitoring 4.8, and SCT switched to that version
